src/uu/dd/src/parseargs.rs

```diff
@@ -474,8 +474,9 @@ fn show_zero_multiplier_warning() {
 }
 
 /// Parse bytes using str::parse, then map error if needed.
-fn parse_bytes_only(s: &str) -> Result<u64, ParseError> {
-    s.parse()
+fn parse_bytes_only(s: &str, i: usize) -> Result<u64, ParseError> {
```

Comment:
What's the reason for the signature change?

Reply:
Because the calling function, which parsed the number/quantifier and called this which returned the incorrect error message, automatically assumed any 'b' at the end stood for byte, stripped it off, and passed on the slice, not the full quantifier. However, GNU prints out the entirety of the quantifier. See the given example in #5905, where GNU prints out `KBb`, but uutils only does `KB`. Because `KB` is the only string passed to this function for that to be used. I'm more than happy to refactor and try out different ways you can suggest, because I realize passing an index might not be the best way (my Rust knowledge isn't good enough to know all the better archaic and obscure things to solve this exact problem haha)
But I don't think the original was correctly architectured to be compatible with GNU in this specific case, so to keep the general architecture, I found passing the calculated index from the calling function to this one, to be used in the error, worked.

Reply:
Ah, I see, that makes sense.